WooCommerce/Classes/ViewRelated/Products/Media/ProductImageActionHandler.swift

```diff
@@ -206,6 +206,18 @@ final class ProductImageActionHandler {
     func resetProductImages(to product: ProductFormDataModel) {
         allStatuses = (productImageStatuses: product.imageStatuses, error: nil)
     }
+
+    /// Updates the product images with the given ones.
+    ///
+    func updateProductImageStatusesAfterReordering(_ productImageStatuses: [ProductImageStatus]) {
+        queue.async { [weak self] in
```

Comment:
> By order do you mean the order of queued operations?

Exactly

> I was asking because I noticed that we're not using `queue.async` when updating statuses in `resetProductImages`. Do you think we should update that as well?

I noticed the same thing during development. IMHO, if we use a queue to modify an array, all operations should be executed by this queue to ensure the consistency of elements. I didn't do it in this task because I thought it was out of the scope of this task. So we could create a new issue or I could modify it in this one 🙂 

WDYT?

Reply:
I think if it's something to make the code safer and doesn't take too much effort it's fine to handle it. Otherwise filing a new issue is also fine.

Reply:
I've found that we only reset the product images when the view disappears, so it's not crucial to use the queue, although I think we should. So I've finally made the change 🙂👍🏼